docs/traffic-control.md

```diff
@@ -0,0 +1,239 @@
+  - [Direction](#direction)
+  - [Action](#action)
+  - [TargetPort](#targetport)
+  - [ReturnPort](#returnport)
+- [Examples](#examples)
+  - [Mirroring all traffic to remote analyzer](#mirroring-all-traffic-to-remote-analyzer)
+  - [Redirecting specific traffic to local analyzer](#redirecting-specific-traffic-to-local-analyzer)
+- [What's next](#whats-next)
+<!-- /toc -->
+
+## What is TrafficControl?
+
+`TrafficControl` is a CRD API that manages and manipulates the transmission of
+Pod traffic. It allows users to mirror or redirect specific traffic originating
+from specific Pods or destined to specific Pods to a local network device or a
+remote destination via various tunnels. It provides full visibility into network
+traffic, including both north-south and east-west traffic.
+
+You may be interested in using this capability if any of the following apply:
+
+- You want to monitor network traffic passing in or out of a set of Pods for
+  purposes such as troubleshooting, intrusion detection, and so on.
+
+- You want to redirect network traffic passing in or out of a set of Pods to
+  applications that analyze traffic, enforce policies, and reject traffic to
+  prevent intrusion.
+
+This guide demonstrates how to configure `TrafficControl` to achieve the above
+result.
+
+## Prerequisites
+
+TrafficControl was introduced in v1.7 as an alpha feature. A feature gate,
+`TrafficControl` must be enabled on the antrea-agent in the `antrea-config`
+ConfigMap like the following options for the feature to work:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    featureGates:
+      TrafficControl: true
+```
+
+## The TrafficControl resource
+
+A TrafficControl in Kubernetes is a REST object. Like all the REST objects, you
+can POST a TrafficControl definition to the API server to create a new instance.
+For example, suppose you have a set of Pods which contain a label `app=web`, the
+following specification creates a new TrafficControl object named
+"mirror-web-app", which mirrors all traffic from or to any Pod with the
+`app=web` label and send them to a collector running on "10.0.10.2" encapsulated
+within VXLAN tunnel:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: mirror-web-app
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        app: web
+  direction: Both
+  action: Mirror
+  targetPort:
+    vxlan:
+      remoteIP: 10.0.10.2
+```
+
+### AppliedTo
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+TrafficControl applies to. Pods can be selected cluster-wide using
+`podSelector`. If set with a `namespaceSelector`, all Pods from Namespaces
+selected by the `namespaceSelector` will be selected. Specific Pods from
+specific Namespaces can be selected by providing both a `podSelector` and a
+`namespaceSelector`. Empty `appliedTo` selects nothing. The field is mandatory.
```

Comment:
IIRC, empty `podSelector` and empty `namespaceSelector` can be used to select all Pods in the cluster?

Reply:
that was also my first thought but I found we made it select nothing from the beginning for NetworkPolicy and Egress so have to keep consistency:
https://github.com/antrea-io/antrea/blob/017f8becc3509d928880069e35418c0956633f70/pkg/controller/grouping/group_entity_index.go#L739
https://github.com/antrea-io/antrea/blob/main/docs/egress.md#appliedto

Reply:
ok, but jut to make sure, the following (which is what I meant by having both selectors be empty) will select all Pods in the cluster:
```yaml
appliedTo:
    namespaceSelector:
    podSelector:
```
while the following (both selectors unset) will select nothing:
```yaml
appliedTo:
```